File: Lab5/Task2/CSharp/LowerCase.cs
using System.Linq;
using System;

namespace CSharp
{
    public class LowerCase : MyString
    {
        public LowerCase(char[] str) : base(str)
        {
            this.Value = str.Select(char.ToLower).ToArray();
        }

        public override void Insert(char elem, int index)
        {
            var newValue = new char[GetLength() + 2];
            Array.Copy(this.Value, 0, newValue, 0, GetLength());
            for (var i = index; i < Value.Length; i++)
            {
                newValue[i + 2] = this.Value[i];
            }
            newValue[index] = '\\';
            newValue[index + 1] = char.ToLower(elem);
            this.Value = newValue;
        }
        protected internal override int GetLength() => base.GetLength();
    }
}

File: Lab5/Task2/C++/C++/LowerCase.cpp
#include "LowerCase.hpp"
#include <cstring>

LowerCase::LowerCase(char* str) : MyString(str) {
    this->Value = new char[strlen(str)];
    for (int i = 0; i < strlen(str); i++){
        this->Value[i] = tolower(str[i]);
    }
}

void LowerCase::Insert(char elem, uint index){
    int length = this->GetLength();
    char* newValue = new char[length + 3];
    for (int i = 0; i < length + 2; i++) {
        if ( i < index ) {
            newValue[i] = Value[i];
        }
        if ( i == index ) {
            newValue[i] = '\\';
        }
        if (i == index + 1){
            newValue[i] = tolower(elem);
        }
        if ( i > index + 1){
            newValue[i] = this->Value[i-2];
        }
    }
    newValue[length+2] = '\0';
    this->Value = newValue;
}

int LowerCase::GetLength() {
    return strlen(this->Value);
}

